quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/storage/infinispan/jgroups/JGroupsConfigurator.java

```diff
@@ -0,0 +1,134 @@
+ */
+public class JGroupsConfigurator {
+
+    private final ConfigurationBuilderHolder holder;
+    private final List<JGroupsStackConfigurator> stackConfiguratorList;
+
+    private JGroupsConfigurator(ConfigurationBuilderHolder holder, List<JGroupsStackConfigurator> stackConfiguratorList) {
+        this.holder = holder;
+        this.stackConfiguratorList = stackConfiguratorList;
+    }
+
+    private static void createJdbcPingConfigurator(ConfigurationBuilderHolder holder, List<JGroupsStackConfigurator> configurator) {
+        var stackXmlAttribute = JGroupsUtil.transportStackOf(holder);
+        if (stackXmlAttribute.isModified() && !isJdbcPingStack(stackXmlAttribute.get())) {
+            CacheManagerFactory.logger.debugf("Custom stack configured (%s). JDBC_PING discovery disabled.", stackXmlAttribute.get());
+            return;
+        }
+        CacheManagerFactory.logger.debug("JDBC_PING discovery enabled.");
+        if (!stackXmlAttribute.isModified()) {
+            // defaults to jdbc-ping
+            JGroupsUtil.transportOf(holder).stack("jdbc-ping");
+        }
+        configurator.add(JGroupsJdbcPingStackConfigurator.INSTANCE);
+    }
+
+    private static boolean isJdbcPingStack(String stackName) {
+        return "jdbc-ping".equals(stackName) || "jdbc-ping-udp".equals(stackName);
+    }
+
+    private static void createTlsConfigurator(List<JGroupsStackConfigurator> configurator) {
+        if (!Configuration.isTrue(CachingOptions.CACHE_EMBEDDED_MTLS_ENABLED)) {
+            CacheManagerFactory.logger.debug("JGroups encryption disabled.");
+            return;
+        }
+
+        if (Configuration.isBlank(CachingOptions.CACHE_EMBEDDED_MTLS_KEYSTORE) && Configuration.isBlank(CachingOptions.CACHE_EMBEDDED_MTLS_TRUSTSTORE)) {
+            CacheManagerFactory.logger.debug("JGroups encryption enabled. Neither KeyStore and Truststore present, using the certificates from database.");
+            configurator.add(JpaJGroupsTlsConfigurator.INSTANCE);
+            return;
+        }
+        CacheManagerFactory.logger.debug("JGroups encryption enabled. KeyStore or Truststore present.");
+        configurator.add(FileJGroupsTlsConfigurator.INSTANCE);
+    }
+
+    private static boolean isLocal(ConfigurationBuilderHolder holder) {
+        return JGroupsUtil.transportOf(holder) == null;
+    }
+
+    public static JGroupsConfigurator create(ConfigurationBuilderHolder holder) {
+        if (InfinispanUtils.isRemoteInfinispan() || isLocal(holder)) {
+            CacheManagerFactory.logger.debug("Multi Site or local mode. Skipping JGroups configuration.");
+            return new JGroupsConfigurator(holder, List.of());
+        }
+        // Configure stack from CLI options to Global Configuration
+        var transportStack = Configuration.getRawValue("kc.cache-stack");
+        if (transportStack != null) {
+            JGroupsUtil.transportOf(holder).stack(transportStack);
+        }
```

Comment:
```suggestion
        Configuration.getOptionalKcValue(CachingOptions.CACHE_STACK).ifPresent(stack -> JGroupsUtil.transportOf(holder).stack(stack));
```
nitpick: Should be sufficient this.